aws_backup_selection.selection: error creating Backup Selection: InvalidParameterValueException

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
0.11.14

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_backup_selection

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
```
       2019-10-15T12:24:03.672+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: Action=DescribeSubnets&SubnetId.1=subnet-0xxxxxx8df&Version=2016-11-15
       2019-10-15T12:24:03.672+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: -----------------------------------------------------
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: 2019/10/15 12:24:03 [DEBUG] [aws-sdk-go] DEBUG: Response Backup/CreateBackupSelection Details:
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: ---[ RESPONSE ]--------------------------------------
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: HTTP/1.1 400 Bad Request
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: Connection: close
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: Content-Length: 295
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: Content-Type: application/json
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: Date: Tue, 15 Oct 2019 10:24:02 GMT
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: X-Amzn-Errortype: InvalidParameterValueException:http://internal.amazon.com/coral/com.amazonaws.services.cryo/
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: X-Amzn-Requestid: ae13be5f-a155-45c3-b886-24c72cbd7fd9
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: -----------------------------------------------------
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: 2019/10/15 12:24:03 [DEBUG] [aws-sdk-go] {"Code":"ERROR_3018","Context":"arn:aws:iam::126666663812:role/ROLE
resname","Message":"IAM Role arn:aws:iam::126666663812:role/ROLEresname is not authorized to call tag:GetResources
","Type":null}
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: 2019/10/15 12:24:03 [DEBUG] [aws-sdk-go] DEBUG: Validate Response Backup/CreateBackupSelection failed, attempt 0/25, e
rror InvalidParameterValueException: IAM Role arn:aws:iam::126666663812:role/ROLEresname is not authorized to call tag:GetResources
       2019-10-15T12:24:03.028+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:   status code: 400, request id: ae13be5f-a155-45c3-b886-24c72cbd7fd9
       2019/10/15 12:24:03 [ERROR] root.mymodule: eval: *terraform.EvalApplyPost, err: 1 error occurred:
        * aws_backup_selection.selection: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::126666663812:role/ROLEresname is not authorized to call tag:GetResources
        status code: 400, request id: ae13be5f-a155-45c3-b886-24c72cbd7fd9

       2019/10/15 12:24:03 [ERROR] root.mymodule: eval: *terraform.EvalSequence, err: 1 error occurred:
        * aws_backup_selection.selection: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::126666663812:role/ROLEresname is not authorized to call tag:GetResources
        status code: 400, request id: ae13be5f-a155-45c3-b886-24c72cbd7fd9

       2019/10/15 12:24:03 [TRACE] [walkApply] Exiting eval tree: module.mymodule.aws_backup_selection.selection

       Error: Error applying plan:

       2019/10/15 12:24:03 [DEBUG] plugin: waiting for all plugin processes to complete...
       1 error occurred:
        * module.mymodule.aws_backup_selection.selection: 1 error occurred:
       2019-10-15T12:24:03.039+0200 [DEBUG] plugin.terraform-provider-random_v2.2.1_x4: 2019/10/15 12:24:03 [ERR] plugin: plugin server: accept unix /tmp/plugin229438604: use of closed network connection
       2019-10-15T12:24:03.039+0200 [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: 2019/10/15 12:24:03 [ERR] plugin: plugin server: accept unix /tmp/plugin054189175: use of closed network connection
        * aws_backup_selection.selection: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::126666663812:role/ROLEresname is not authorized to call tag:GetResources
        status code: 400, request id: ae13be5f-a155-45c3-b886-24c72cbd7fd9


```


### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
N/A
### Expected Behavior

It should retry.
My code doesn't fail always, but sometimes :-|

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->
It Errors out after first try (as per my understanding).

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #9297

I've created the following `main.tf` file based on terraform default documentation:
```
$ cat main.tf
provider "random" {
  version = "2.0.0"
}

resource "random_id" "id" {
  keepers {
    lifecycle = "${terraform.workspace}"
  }

  byte_length = 4
}

locals {
  id = "${random_id.id.hex}"
}


resource "aws_iam_role" "example" {
  name               = "example-${local.id}"
  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": ["sts:AssumeRole"],
      "Effect": "allow",
      "Principal": {
        "Service": ["backup.amazonaws.com"]
      }
    }
  ]
}
POLICY
}

resource "aws_iam_role_policy_attachment" "example" {
  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
  role       = "${aws_iam_role.example.name}"
}

resource "aws_backup_plan" "example" {
  name = "tf_example_backup_plan-${local.id}"

  rule {
    rule_name         = "tf_example_backup_rule"
    target_vault_name = "${aws_backup_vault.example.name}"
    schedule          = "cron(0 12 * * ? *)"
  }
}

resource "aws_backup_selection" "example" {
  iam_role_arn = "${aws_iam_role.example.arn}"
  name         = "tf_example_backup_selection-${local.id}"
  plan_id      = "${aws_backup_plan.example.id}"

  selection_tag {
    type  = "STRINGEQUALS"
    key   = "foo"
    value = "bar"
  }
}

resource "aws_backup_vault" "example" {
  name        = "example_backup_vault-${local.id}"
  #kms_key_arn = "${aws_kms_key.example.arn}"
}

```


This code is executed in a loop until the error is raised.
 
The code was applied successfully for 3 times and failed after - see attached (encrypted logs).


```
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4:
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4:
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4: -----------------------------------------------------
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4: 2019/10/19 19:12:36 [DEBUG] [aws-sdk-go] {"Code":"ERROR_3018","Context":"arn:aws:iam::121212121212:role/example-67c8a75a","Message":"IAM Role arn:aws:iam::121212121212:role/example-67c8a75a is not authorized to call tag:GetResources","Type":null}
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4: 2019/10/19 19:12:36 [DEBUG] [aws-sdk-go] DEBUG: Validate Response Backup/CreateBackupSelection failed, attempt 0/25, error InvalidParameterValueException: IAM Role arn:aws:iam::121212121212:role/example-67c8a75a is not authorized to call tag:GetResources
2019-10-19T19:12:36.721+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4:  status code: 400, request id: 0be67a96-44e1-43ba-9e33-efdd82a4faa0
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalWriteState
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalIf
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalWriteState
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalWriteDiff
2019/10/19 19:12:36 [TRACE] root: eval: *terraform.EvalApplyPost
2019/10/19 19:12:36 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error occurred:
        * aws_backup_selection.example: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::121212121212:role/example-67c8a75a is not authorized to call tag:GetResources
        status code: 400, request id: 0be67a96-44e1-43ba-9e33-efdd82a4faa0

2019/10/19 19:12:36 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error occurred:
        * aws_backup_selection.example: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::121212121212:role/example-67c8a75a is not authorized to call tag:GetResources
        status code: 400, request id: 0be67a96-44e1-43ba-9e33-efdd82a4faa0

2019/10/19 19:12:36 [TRACE] [walkApply] Exiting eval tree: aws_backup_selection.example
2019/10/19 19:12:36 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/10/19 19:12:36 [TRACE] dag/walk: upstream errored, not walking "provider.aws (close)"
2019/10/19 19:12:36 [TRACE] dag/walk: upstream errored, not walking "root"
2019/10/19 19:12:36 [TRACE] Preserving existing state lineage "933bb3ad-332b-f487-f8f3-feb0201354ac"
2019/10/19 19:12:36 [TRACE] Preserving existing state lineage "933bb3ad-332b-f487-f8f3-feb0201354ac"
2019/10/19 19:12:36 [TRACE] Preserving existing state lineage "933bb3ad-332b-f487-f8f3-feb0201354ac"
2019/10/19 19:12:36 [DEBUG] plugin: waiting for all plugin processes to complete...
ESC[31m
ESC[1mESC[31mError: ESC[0mESC[0mESC[1mError applying plan:

1 error occurred:
        * aws_backup_selection.example: 1 error occurred:
        * aws_backup_selection.example: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::121212121212:role/example-67c8a75a is not authorized to call tag:GetResources
2019-10-19T19:12:36.728+0200 [DEBUG] plugin.terraform-provider-aws_v2.33.0_x4: 2019/10/19 19:12:36 [ERR] plugin: plugin server: accept unix /tmp/plugin933888988: use of closed network connection
        status code: 400, request id: 0be67a96-44e1-43ba-9e33-efdd82a4faa0





Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.ESC[0m


```


[logging-tf-apply.20191019T1910.log.gpg.gz](https://github.com/terraform-providers/terraform-provider-aws/files/3746982/logging-tf-apply.20191019T1910.log.gpg.gz)

--- a/aws/resource_aws_backup_selection.go
+++ b/aws/resource_aws_backup_selection.go
@@ -105,14 +105,14 @@ func resourceAwsBackupSelectionCreate(d *schema.ResourceData, meta interface{}) 
 		// Retry on the following error:
 		// InvalidParameterValueException: IAM Role arn:aws:iam::123456789012:role/XXX cannot be assumed by AWS Backup
 		if isAWSErr(err, backup.ErrCodeInvalidParameterValueException, "cannot be assumed") {
-                        log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
+			log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
 			return resource.RetryableError(err)
 		}
 
 		// Retry on the following error:
 		// InvalidParameterValueException: IAM Role arn:aws:iam::123456789012:role/XXX is not authorized to call tag:GetResources
 		if isAWSErr(err, backup.ErrCodeInvalidParameterValueException, "is not authorized to call") {
-                        log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
+			log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
 			return resource.RetryableError(err)
 		}
 

--- a/aws/resource_aws_backup_selection.go
+++ b/aws/resource_aws_backup_selection.go
@@ -105,6 +105,14 @@ func resourceAwsBackupSelectionCreate(d *schema.ResourceData, meta interface{}) 
 		// Retry on the following error:
 		// InvalidParameterValueException: IAM Role arn:aws:iam::123456789012:role/XXX cannot be assumed by AWS Backup
 		if isAWSErr(err, backup.ErrCodeInvalidParameterValueException, "cannot be assumed") {
+                        log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
+			return resource.RetryableError(err)
+		}
+
+		// Retry on the following error:
+		// InvalidParameterValueException: IAM Role arn:aws:iam::123456789012:role/XXX is not authorized to call tag:GetResources
+		if isAWSErr(err, backup.ErrCodeInvalidParameterValueException, "is not authorized to call") {
+                        log.Printf("[DEBUG] Received %s, retrying create backup selection.", err)
 			return resource.RetryableError(err)
 		}
 

